sys/include/psa/crypto_sizes.h

```diff
@@ -0,0 +1,1001 @@
+ *          @ref psa_mac_verify_finish(), for any of the supported key types and MAC algorithms.
+ *
+ * @details If the size of the MAC buffer is at least this large, it is guaranteed that
+ *          @ref psa_mac_verify() and @ref psa_mac_verify_finish() will not fail due to an
+ *          insufficient buffer size.
+ *
+ *          See also @ref PSA_MAC_LENGTH().
+ */
+#define PSA_MAC_MAX_SIZE (PSA_HASH_MAX_SIZE)
+
+/**
+ * @brief   The block size of a block cipher.
+ *
+ * @note    It is possible to build stream cipher algorithms on top of a block cipher,
+ *          for example CTR mode (@ref PSA_ALG_CTR). This macro only takes the key type
+ *          into account, so it cannot be used to determine the size of the data that
+ *          @ref psa_cipher_update() might buffer for future processing in general.
+ *
+ * @param   type  A cipher key type (value of type @ref psa_key_type_t).
+ *
+ * @return  The block size for a block cipher, or 1 for a stream cipher.
+ */
+#define PSA_BLOCK_CIPHER_BLOCK_LENGTH(type)                                     \
+    (1u << (((type) >> 8) & 7))
+
+/**
+ * @brief   The maximum block size of a block cipher supported by the implementation.
+ *
+ * @details See also @ref PSA_BLOCK_CIPHER_BLOCK_LENGTH().
+ */
+#define PSA_BLOCK_CIPHER_BLOCK_MAX_SIZE /* implementation-defined value */
+
+/**
+ * @brief   A sufficient output buffer size for @ref psa_cipher_decrypt(), for any of the supported
+ *          key types and cipher algorithms.
+ *
+ * @details If the size of the output buffer is at least this large, it is guaranteed that
+ *          @ref psa_cipher_decrypt() will not fail due to an insufficient buffer size.
+ *
+ *          See also @ref PSA_CIPHER_DECRYPT_OUTPUT_SIZE().
+ *
+ * @param   input_length Size of the input in bytes.
+ */
+#define PSA_CIPHER_DECRYPT_OUTPUT_MAX_SIZE(input_length) \
+/* implementation-defined value */
+
+/**
+ * @brief   The maximum size of the output of @ref psa_cipher_decrypt(), in bytes.
+ *
+ * @details If the size of the output buffer is at least this large, it is guaranteed
+ *          that @ref psa_cipher_decrypt() will not fail due to an insufficient buffer size.
+ *          Depending on the algorithm, the actual size of the output might be smaller.
+ *
+ *          See also @ref PSA_CIPHER_DECRYPT_OUTPUT_MAX_SIZE.
+ *
+ * @param   key_type        A symmetric key type that is compatible with algorithm alg.
+ * @param   alg             A cipher algorithm (PSA_ALG_XXX value such that
+ *                          @ref PSA_ALG_IS_CIPHER(@p alg) is true).
+ * @param   input_length    Size of the input in bytes.
+ *
+ * @return  A sufficient output size for the specified key type and algorithm.
+ *          0 if the key type or cipher algorithm is not recognized, or the parameters
+ *          are incompatible.
+ */
+#define PSA_CIPHER_DECRYPT_OUTPUT_SIZE(key_type, alg, input_length) \
```

Comment:
Would be good if this macro considered that for e.g. `AES_CBC`  the output size does not contain the IV anymore. Makes using the API easier.

Reply:
Fixed